src/postgres.c

```diff
@@ -140,7 +140,15 @@ postgres_producer_produce(Producer p, Message msg)
     Meta m = (Meta)p->meta;
 
     char *buf = (char *) message_get_data(msg);
+    size_t len = message_get_len(msg);
     char *newline = "\n";
+
+    if (buf[len] != '\0')
```

Comment:
`memrchr(buf, 0, len)` is an alternative to consider.

Reply:
nice suggestion, but that is _GNU_SOURCE